src/mod_register.erl

```diff
@@ -196,20 +225,21 @@ get_username_and_password_values(Q) ->
     {exml_query:path(Q, [{element, <<"username">>}, cdata]),
      exml_query:path(Q, [{element, <<"password">>}, cdata])}.
 
-register_or_change_password(Credentials, ClientJID, #jid{lserver = ServerDomain}, IQ, IPAddr) ->
+register_or_change_password(HostType, Credentials, ClientJID, #jid{lserver = ServerDomain}, IQ, IPAddr) ->
     {Username, Password} = Credentials,
-    case inband_registration_and_cancelation_allowed(ServerDomain, ClientJID) of
+    case inband_registration_and_cancelation_allowed(HostType, ServerDomain, ClientJID) of
         true ->
             #iq{sub_el = Children, lang = Lang} = IQ,
-            try_register_or_set_password(Username, ServerDomain, Password,
+            try_register_or_set_password(HostType, Username, ServerDomain, Password,
                                          ClientJID, IQ, Children, IPAddr, Lang);
         false ->
             %% This is not described in XEP 0077.
             error_response(IQ, mongoose_xmpp_errors:forbidden())
     end.
 
-attempt_cancelation(#jid{} = ClientJID, #jid{lserver = ServerDomain}, #iq{} = IQ) ->
-    case inband_registration_and_cancelation_allowed(ServerDomain, ClientJID) of
+attempt_cancelation(HostType, #jid{} = ClientJID, #jid{lserver = ServerDomain}, #iq{} = IQ) ->
+    %Is replacing server lserver with passed hostname ok?  
```

Comment:
This is some TODO or just forgot remove comments?

Reply:
Yea forgot. Sorry